### PR TITLE
Add helm charts of newspaper api

### PR DESCRIPTION
**Pull request by gollariel**

### CODE REVIEW

**Comment by gollariel** [comment]
@sergey-rozin-smarp Please, let me know if I miss something. Also, I did not test it, I would like to ask for help with testing

File: helm-chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "helm-chart.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "helm-chart.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (sergey-rozin-smarp) [inline helm-chart/templates/_helpers.tpl:50]: Oriol might ask you to add
`app: {{ include "helm-chart.name" . }}`
This label is used just for similarity to our current deployments. Also it's interesting why your chart is using "helm-chart" as a base object. May be you used something like "helm create helm-chart" to generate it. I'd prefer to have actual app name here but it's not critical

File: helm-chart/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
```
Comment (sergey-rozin-smarp) [inline helm-chart/templates/deployment.yaml:40]: are these probes correct? Empty path looks suspicious

File: helm-chart/templates/hpa.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
```
Comment (sergey-rozin-smarp) [inline helm-chart/templates/hpa.yaml:2]: update it to `autoscaling/v2beta2`. You can use slackapp chart as an example

File: helm-chart/templates/deployment.yaml (hunk 2 above)
Comment (sergey-rozin-smarp) [reply helm-chart/templates/deployment.yaml:40]: in values.yaml you define probe paths but they are not used here

**Comment by sergey-rozin-smarp** [comment]
what about using this helm chart on CI?

File: helm-chart/templates/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+{{- if .Values.ingress.enabled -}}
```
Comment (orifito) [inline helm-chart/templates/ingress.yaml:1]: this whole template can be removed.. and the `NOTES.txt` will need to be updated accordingly

File: helm-chart/values.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
```
Comment (orifito) [inline helm-chart/values.yaml:61]: ditto, this can be removed

File: helm-chart/values.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious
```
Comment (orifito) [inline helm-chart/values.yaml:64]: please set current resources configured:
```
    Limits:
      cpu:     750m
      memory:  640Mi
    Requests:
      cpu:      400m
      memory:   400Mi
```

File: newspaper-api.yml [hunk 7]
```diff
@@ -51,7 +51,6 @@ spec:
       containers:
```
Comment (orifito) [inline newspaper-api.yml:51]: this file should be removed too

**Review by orifito — CHANGES_REQUESTED** [review]

File: helm-chart/values.yaml [hunk 8]
```diff
@@ -44,33 +44,17 @@ service:
...
   # choice for the user. This also increases chances charts run on environments with little
   # resources, such as Minikube. If you do want to specify resources, uncomment the following
   # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-  # limits:
-  #   cpu: 100m
-  #   memory: 128Mi
-  # requests:
-  #   cpu: 100m
-  #   memory: 128Mi
+   limits:
```
Comment (orifito) [inline helm-chart/values.yaml:52]: bad indentation here, there's an extra space

File: helm-chart/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            httpGet:
+              path: {{ .Values.application.livenessProbe.path }}
+              port: {{ .Values.application.livenessProbe.port }}
```
Comment (orifito) [inline helm-chart/templates/deployment.yaml:42]: this is not set / present on the values file

File: helm-chart/values.yaml [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  # It's required to set frontendPath for each deployment
+  frontendPath:
+  apiPort: 38765
+  livenessProbe:
+    path: /health
+  readinessProbe:
+    path: /health
+
+image:
+  repository: smarp/newspaper-api:v2.4
```
Comment (orifito) [inline helm-chart/values.yaml:13]: remove tag from there since it's set in the key below

**Review by orifito — CHANGES_REQUESTED** [review]

**Review by orifito — CHANGES_REQUESTED** [review]
ConfigMap is missing (see last lines of `newspaper-api.yaml`)

**Review by orifito — APPROVED** [review]

**Pull request merged**